[BUG] truncate function throws error when formatting table output

### Describe the bug

- [x] Read the [CONTRIBUTING.md](../blob/trunk/.github/CONTRIBUTING.md)
- [x] Checked for existing issues to verify that the bug has not already been submitted.

The `truncate` function for formatting text within a table seems to be broken for ~~empty strings~~ `null` values.
```zsh
gh --version
gh version 2.10.1 (2022-05-10)
https://github.com/cli/cli/releases/tag/v2.10.1
```
### Steps to reproduce the behavior

1. Star a repo that has no description.
2. Run the following command to list your starred repos.
```zsh
gh api user/starred --template '{{range .}}{{tablerow "⭐" (.description|truncate 15) }}{{end}}'
```
3. `truncate` function demands a `string`, see the 2nd example below.

### Logs

```zsh
❯ gh api user/starred --template '{{range .}}{{tablerow "⭐" .description }}{{end}}'
⭐  Night theme for Zotero UI and PDF
⭐  🐸💬 - a deep learning toolkit for Text-to-Speech, battle-tested in research and production
⭐  Generates cheatsheets for your (favourite) apps
⭐  macOS Scripting for Configuration, Backup and Restore
⭐
⭐  Technical infos of open source games.

❯ gh api user/starred --template '{{range .}}{{tablerow "⭐" (.description|truncate 15) }}{{end}}'
template: :1:51: executing "" at <15>: invalid value; expected string
```

### Related
Issue #3488

Kindly pinging @heaths

Thank you for your reply. I will do my best to provide enough information to find the cause of the problem.
I am using `zsh` (macOS 10.15.7), see image below for more clarity.
<img src="https://ttm.sh/bsT.52.jpg" width="400">

Switching to the default login shell made no difference.

## Running tests with a modified file
I consistently fail when I change the following lines in  [template_test.go](../blob/trunk/pkg/export/template_test.go#L190-L197)
```go
			name: "table with color",
			args: args{
				json: strings.NewReader(heredoc.Doc(`[
					{"number": 1, "title": null}
				]`)),
				template: `{{range .}}{{tablerow (.number | color "green") (.title | truncate 2)}}{{end}}`,
			},
			wantW: "\x1b[0;32m1\x1b[0m  \n",
```
### Log output for testing
```zsh
❯ go test ./...
ok      github.com/cli/cli/v2/api       (cached)
ok      github.com/cli/cli/v2/cmd/gen-docs      (cached)
ok      github.com/cli/cli/v2/cmd/gh    (cached)
ok      github.com/cli/cli/v2/context   (cached)
ok      github.com/cli/cli/v2/git       (cached)
?       github.com/cli/cli/v2/internal/authflow [no test files]
?       github.com/cli/cli/v2/internal/build    [no test files]
ok      github.com/cli/cli/v2/internal/codespaces       (cached)
ok      github.com/cli/cli/v2/internal/codespaces/api   (cached)
ok      github.com/cli/cli/v2/internal/config   (cached)
ok      github.com/cli/cli/v2/internal/docs     (cached)
ok      github.com/cli/cli/v2/internal/ghinstance       (cached)
ok      github.com/cli/cli/v2/internal/ghrepo   (cached)
?       github.com/cli/cli/v2/internal/httpunix [no test files]
?       github.com/cli/cli/v2/internal/run      [no test files]
ok      github.com/cli/cli/v2/internal/update   (cached)
?       github.com/cli/cli/v2/pkg/cmd/actions   [no test files]
?       github.com/cli/cli/v2/pkg/cmd/alias     [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/alias/delete      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/alias/expand      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/alias/list        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/alias/set (cached)
ok      github.com/cli/cli/v2/pkg/cmd/api       (cached)
?       github.com/cli/cli/v2/pkg/cmd/auth      [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/auth/gitcredential        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/login        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/logout       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/refresh      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/setupgit     (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/shared       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/auth/status       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/browse    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/codespace (cached)
ok      github.com/cli/cli/v2/pkg/cmd/completion        (cached)
?       github.com/cli/cli/v2/pkg/cmd/config    [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/config/get        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/config/list       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/config/set        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/extension (cached)
ok      github.com/cli/cli/v2/pkg/cmd/factory   (cached)
?       github.com/cli/cli/v2/pkg/cmd/gist      [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/gist/clone        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/create       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/delete       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/edit (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/list (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/shared       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gist/view (cached)
?       github.com/cli/cli/v2/pkg/cmd/gpg-key   [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/gpg-key/add       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/gpg-key/list      (cached)
?       github.com/cli/cli/v2/pkg/cmd/issue     [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/issue/close       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/comment     (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/create      2.590s
ok      github.com/cli/cli/v2/pkg/cmd/issue/delete      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/edit        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/list        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/reopen      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/shared      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/status      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/transfer    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/issue/view        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/label     (cached)
?       github.com/cli/cli/v2/pkg/cmd/pr        [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/pr/checkout       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/checks (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/close  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/comment        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/create (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/diff   (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/edit   (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/list   (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/merge  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/ready  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/reopen (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/review (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/shared (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/status (cached)
ok      github.com/cli/cli/v2/pkg/cmd/pr/view   (cached)
?       github.com/cli/cli/v2/pkg/cmd/release   [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/release/create    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/delete    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/delete-asset      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/download  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/edit      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/list      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/release/shared    (cached)
?       github.com/cli/cli/v2/pkg/cmd/release/upload    [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/release/view      (cached)
?       github.com/cli/cli/v2/pkg/cmd/repo      [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/repo/archive      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/clone        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/create       (cached)
?       github.com/cli/cli/v2/pkg/cmd/repo/credits      [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/repo/delete       (cached)
?       github.com/cli/cli/v2/pkg/cmd/repo/deploy-key   [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/repo/deploy-key/add       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/deploy-key/delete    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/deploy-key/list      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/edit (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/fork (cached)
?       github.com/cli/cli/v2/pkg/cmd/repo/garden       [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/repo/list (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/rename       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/sync (cached)
ok      github.com/cli/cli/v2/pkg/cmd/repo/view (cached)
ok      github.com/cli/cli/v2/pkg/cmd/root      (cached)
?       github.com/cli/cli/v2/pkg/cmd/run       [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/run/cancel        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/download      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/list  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/rerun (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/shared        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/view  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/run/watch (cached)
?       github.com/cli/cli/v2/pkg/cmd/search    [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/search/issues     (cached)
ok      github.com/cli/cli/v2/pkg/cmd/search/prs        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/search/repos      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/search/shared     (cached)
?       github.com/cli/cli/v2/pkg/cmd/secret    [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/secret/list       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/secret/remove     (cached)
ok      github.com/cli/cli/v2/pkg/cmd/secret/set        (cached)
ok      github.com/cli/cli/v2/pkg/cmd/secret/shared     (cached)
?       github.com/cli/cli/v2/pkg/cmd/ssh-key   [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/ssh-key/add       (cached)
ok      github.com/cli/cli/v2/pkg/cmd/ssh-key/list      (cached)
ok      github.com/cli/cli/v2/pkg/cmd/status    (cached)
ok      github.com/cli/cli/v2/pkg/cmd/version   (cached)
?       github.com/cli/cli/v2/pkg/cmd/workflow  [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/workflow/disable  (cached)
ok      github.com/cli/cli/v2/pkg/cmd/workflow/enable   (cached)
ok      github.com/cli/cli/v2/pkg/cmd/workflow/list     (cached)
ok      github.com/cli/cli/v2/pkg/cmd/workflow/run      (cached)
?       github.com/cli/cli/v2/pkg/cmd/workflow/shared   [no test files]
ok      github.com/cli/cli/v2/pkg/cmd/workflow/view     (cached)
ok      github.com/cli/cli/v2/pkg/cmdutil       (cached)
--- FAIL: Test_executeTemplate (0.00s)
    --- FAIL: Test_executeTemplate/table_with_color (0.00s)
        template_test.go:272: executeTemplate() error = template: :1:67: executing "" at <2>: invalid value; expected string, wantErr false
FAIL
FAIL    github.com/cli/cli/v2/pkg/export        3.088s
?       github.com/cli/cli/v2/pkg/extensions    [no test files]
?       github.com/cli/cli/v2/pkg/findsh        [no test files]
ok      github.com/cli/cli/v2/pkg/githubtemplate        (cached)
?       github.com/cli/cli/v2/pkg/httpmock      [no test files]
ok      github.com/cli/cli/v2/pkg/iostreams     (cached)
ok      github.com/cli/cli/v2/pkg/jsoncolor     (cached)
ok      github.com/cli/cli/v2/pkg/liveshare     (cached)
?       github.com/cli/cli/v2/pkg/liveshare/test        [no test files]
ok      github.com/cli/cli/v2/pkg/markdown      (cached)
?       github.com/cli/cli/v2/pkg/prompt        [no test files]
ok      github.com/cli/cli/v2/pkg/search        (cached)
ok      github.com/cli/cli/v2/pkg/set   (cached)
ok      github.com/cli/cli/v2/pkg/surveyext     (cached)
ok      github.com/cli/cli/v2/pkg/text  (cached)
?       github.com/cli/cli/v2/script    [no test files]
?       github.com/cli/cli/v2/test      [no test files]
ok      github.com/cli/cli/v2/utils     (cached)
FAIL

╭─ ~/Desktop/cli trunk *
╰─❯
```

--- a/pkg/export/template.go
+++ b/pkg/export/template.go
@@ -56,7 +56,15 @@ func (t *Template) parseTemplate(tpl string) (*template.Template, error) {
 		"join":        templateJoin,
 		"tablerow":    t.tableRow,
 		"tablerender": t.tableRender,
-		"truncate":    text.Truncate,
+		"truncate": func(maxWidth int, v interface{}) (string, error) {
+			if v == nil {
+				return "", nil
+			}
+			if s, ok := v.(string); ok {
+				return text.Truncate(maxWidth, s), nil
+			}
+			return "", fmt.Errorf("invalid value; expected string, got %T", v)
+		},
 	}
 
 	if !t.io.ColorEnabled() {

--- a/pkg/export/template_test.go
+++ b/pkg/export/template_test.go
@@ -263,6 +263,38 @@ func Test_executeTemplate(t *testing.T) {
 			},
 			wantW: "This is a ...",
 		},
+		{
+			name: "truncate with JSON null",
+			args: args{
+				json:     strings.NewReader(`{}`),
+				template: `{{ truncate 13 .title }}`,
+			},
+			wantW: "",
+		},
+		{
+			name: "truncate with piped JSON null",
+			args: args{
+				json:     strings.NewReader(`{}`),
+				template: `{{ .title | truncate 13 }}`,
+			},
+			wantW: "",
+		},
+		{
+			name: "truncate with piped JSON null in parenthetical",
+			args: args{
+				json:     strings.NewReader(`{}`),
+				template: `{{ (.title | truncate 13) }}`,
+			},
+			wantW: "",
+		},
+		{
+			name: "truncate invalid type",
+			args: args{
+				json:     strings.NewReader(`{"title": 42}`),
+				template: `{{ (.title | truncate 13) }}`,
+			},
+			wantErr: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

